packages/asset-swapper/src/utils/market_operation_utils/bancor_service.ts

```diff
@@ -0,0 +1,36 @@
+import { BigNumber } from '@0x/utils';
+import { SDK } from '@bancor/sdk';
+import { BlockchainType, Token } from '@bancor/sdk/dist/types';
+
+import { BancorQuoteData } from './types';
+
+function token(address: string, blockchainType: BlockchainType = BlockchainType.Ethereum): Token {
+    return {
+        blockchainType,
+        blockchainId: address,
+    };
+}
+
+export class BancorService {
+    // Bancor recommends setting this value to 2% under the expected return amount
+    public minReturnAmountBufferPercentage = 0.98;
+    private _sdk?: SDK;
+
+    constructor(public ethereumNodeEndpoint: string) {}
+
+    public async getSDKAsync(): Promise<SDK> {
+        if (!this._sdk) {
+            this._sdk = await SDK.create({ ethereumNodeEndpoint: this.ethereumNodeEndpoint });
```

Comment:
is there a way to give it a `provider` instead of having to pass down the RPC url?

Reply:
Haven't found one unfortunately :/

Reply:
It looks like you can pass in a web3 provider, might try just wrapping our provider in something like `{ sendAsync(payload) => provider.send(payload) }` or whatever works. https://github.com/bancorprotocol/sdk/blob/master/src/types.ts

Reply:
yeaaa it works! sorry I missed it on the first go round.